### PR TITLE
docs: Add example for Google Workflow Identity Federation

### LINKED ISSUE #745: Add example for Google Workflow Identity Federation
We probably want to add an example where Google WIF is used, not only what we got.

Should probably land #743 first

### PR DESCRIPTION
**Pull request by cobraz**
fixes #745

### CODE REVIEW

File: examples/nodejs-google-workload-identity-federation.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ secrets.SERVICE_ACCOUNT_EMAIL }}
+
+      - name: Applying infrastructure 🚀
+        uses: pulumi/actions@v3
+        with:
+          command: up
+          stack-name: dev
+          cloud-url: gs://my-bucket-name
```
[the remark this answers is not in the document]
Comment (cobraz) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: Sure, but I feel that falls outside of this example, right? Users might even use Google KMS, which would move this configuration outside of this example.

[the remark this answers is not in the document]
Comment (Frassle) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: If they we're they'd still need to set secrets-provider in the action config. There's no environment in which this example would work, and that feels like a bad example.

[the remark this answers is not in the document]
Comment (cobraz) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: Isn't that normal?

[the remark this answers is not in the document]
Comment (Frassle) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: Eh although seems we don't do this for _any_ other examples so 🤷‍♂️ maybe its not imporant.

[the remark this answers is not in the document]
Comment (cobraz) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: I can only speak for what I've seen out in the wild, which is that most use Google KMS.

Maybe we can cover both areas with a simple comment?

```suggestion
          cloud-url: gs://my-bucket-name
        # If you're not using Google KMS, you probably need to PULUMI_CONFIG_PASSPHRASE
        # 
        # example:
        # env:
        #  PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
```

Something like that?

File: examples/nodejs-google-workload-identity-federation.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+on:
+  push:
+    branches:
+      - main
+jobs:
+  up:
+    name: Update
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎️
```
Comment (RobbieMcKinstry) [inline examples/nodejs-google-workload-identity-federation.yaml:14]: Love the emojis!

File: examples/nodejs-google-workload-identity-federation.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (RobbieMcKinstry) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: I don't have a ton of context on this backend, so forgive me if my response isn't apropos.

1. I think our examples should aim to be _complete_ and _minimal_. That is, they should contain all of the necessary configuration, and should only include code necessary to convey the minimal example. In this case, I think that means assuming Google Blob Storage is used for state management while the Pulumi Service is used for secret management. Assuming they use Google KMS for secret management violates the goal of minimalism, and eliding the `PULUMI_ACCESS_TOKEN` would violate _completion_.

2. As such, I'd suggest the inverse of @cobraz's suggestion above:

```yaml
# You can elide PULUMI_ACCESS_TOKEN below if you're using Google KMS for secret encryption.
env:
        PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
```
 I'd include the access token, and add a comment saying you can remove it if you set up Google KMS.

**Review by RobbieMcKinstry — CHANGES_REQUESTED** [review]
I left a comment weighing in on the discussion above. Otherwise, the example looks good to me! :D  

Feel free to re-request review whenever you want :)

File: examples/nodejs-google-workload-identity-federation.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Frassle) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: > I think that means assuming Google Blob Storage is used for state management while the Pulumi Service is used for secret management

Just an FYI that that is not supported. Service secrets are tied to stacks.

[the remark this answers is not in the document]
Comment (cobraz) [reply examples/nodejs-google-workload-identity-federation.yaml:37]: How about this?

[`9b4b5aa` (#747)](https://github.com/pulumi/actions/pull/747/commits/9b4b5aafb2c567fa7b7fdf86ed08a831e5e3872b)

**Review by RobbieMcKinstry — APPROVED** [review]

**Comment by RobbieMcKinstry** [comment]
Looks like @Frassle concern has been resolved; I'm going to merge. We can always back it out of follow-on if there's an issue.

**Pull request merged**